crypto/src/key/hdkd/derivation_path.rs

```diff
@@ -16,29 +16,71 @@
 use super::child_number::ChildNumber;
 use super::derivable::DerivationError;
 use core::fmt;
+use serialization::{Decode, Encode, Error, Input, Output};
 use std::fmt::Formatter;
 use std::str::FromStr;
 
 const PREFIX: &str = "m";
 const SEPARATOR: &str = "/";
+/// The typical path size in BIP44 is 5 and we add this path limit
+/// in order to support SLIP32 key serialization
+const MAX_PATH_SIZE: usize = u8::MAX as usize;
 
 /// BIP-32 compatible derivation path
 #[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd)]
 pub struct DerivationPath(Vec<ChildNumber>);
 
 impl DerivationPath {
-    pub fn push(&mut self, num: ChildNumber) {
-        self.0.push(num);
+    pub fn empty() -> Self {
+        DerivationPath(vec![])
     }
 
-    pub fn pop(&mut self) -> Option<ChildNumber> {
-        self.0.pop()
+    pub fn into_vec(self) -> Vec<ChildNumber> {
+        self.0
+    }
+
+    pub fn len(&self) -> usize {
+        self.0.len()
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+}
+
+impl Encode for DerivationPath {
+    fn encode_to<T: Output + ?Sized>(&self, dest: &mut T) {
+        let size = self.0.len();
+        debug_assert!(size <= MAX_PATH_SIZE);
+        let size = size as u8;
+        dest.push_byte(size);
+        for num in &self.0 {
+            dest.write(&num.into_encoded_be_bytes());
+        }
     }
 }
 
-impl From<Vec<ChildNumber>> for DerivationPath {
-    fn from(path: Vec<ChildNumber>) -> Self {
-        DerivationPath(path)
+impl Decode for DerivationPath {
+    fn decode<I: Input>(input: &mut I) -> Result<Self, Error> {
+        let size = input.read_byte()?;
+        let mut path = Vec::with_capacity(size as usize);
+
+        for _ in 0..size {
+            path.push(<ChildNumber>::decode(input)?);
+        }
```

Comment:
If we're to use the default encoding underneath, I actually would wonder why we don't use `WrapperTypeEncode` and `WrapperTypeDecode`.

Reply:
> This custom encoding is based on the [SLIP32 specification](https://github.com/satoshilabs/slips/blob/master/slip-0032.md). I recommend sticking with this one so that third party tools can easily support the Mintlayer exported keys.

That sounds sensible to me.